borsh-derive/src/internals/deserialize/enums/mod.rs

```diff
@@ -2,17 +2,17 @@ use proc_macro2::TokenStream as TokenStream2;
 use quote::quote;
 use syn::{Fields, Ident, ItemEnum, Path, WhereClause};
 
-use crate::{
-    attribute_helpers::{
-        contains_initialize_with, contains_skip, contains_use_discriminant, field, BoundType,
-    },
+use crate::internals::{
+    attributes::field::contains_skip,
+    attributes::item::{contains_initialize_with, contains_use_discriminant},
+    attributes::{field, BoundType},
+    deserialize::field_deserialization_output,
     enum_discriminant_map::discriminant_map,
     generics::{compute_predicates, without_defaults, FindTyParams},
-    struct_de::field_deserialization_output,
 };
```

Comment:
These imports in this and other similar files can further be streamlined by replacing this to:
 
```rust
use super::{
    field,
    item,
    BoundType,
    field_deserialization_output,
    discriminant_map,
    generics,
};
```
adding dispatch of imports in `borsh-derive/src/internals/deserialize/mod.rs`

```rust
use crate::internals::attributes::{field, item, BoundType};
use crate::internals::enum_discriminant_map::discriminant_map;
use crate::internals::generics;
```

and using (a little bit more) fully qualified names below:
```rust
...
    let use_discriminant = item::contains_use_discriminant(input)?;
...
    let mut deserialize_params_visitor = generics::FindTyParams::new(&generics);
...
    let de_predicates = generics::compute_predicates(
        deserialize_params_visitor.process_for_bounds(),
        &de_trait_path,
    );
...
```
which is imho more readable than it is, as it adds some context to specific funcs